### PR TITLE
Remove phone appointment feature flag

### LINKED ISSUE #43557: Remove and Clean up Telephone Appointments Feature Flag
## Tasks
- [ ] Remove and Clean up Telephone Appointments Feature Flag

### PR DESCRIPTION
**Pull request by wullaski**
## Description

- Removes the feature flag 
- Cleans up code blocks that use it
- Removes related tests
- Renames AppointmentBlockWithIcons to AppointmentBlock

## Original issue(s)
[department-of-veterans-affairs/va.gov-team#43557](https://github.com/department-of-veterans-affairs/va.gov-team/issues/43557)

### CODE REVIEW

**Review by it-harrison — APPROVED** [review]

File: src/applications/check-in/components/tests/AppointmentBlock.unit.spec.jsx
Comment (brianseek) [inline src/applications/check-in/components/tests/AppointmentBlock.unit.spec.jsx:197]: this test should probably be moved out of the phone context block.

**Review by brianseek — CHANGES_REQUESTED** [review]
One minor thing in a test to move. Awesome work! This will be so nice once it is merged.
If this gets approved later, make sure to not merge it before its blocker is resolved.

File: src/applications/check-in/components/tests/AppointmentBlock.unit.spec.jsx
Comment (wullaski) [reply src/applications/check-in/components/tests/AppointmentBlock.unit.spec.jsx]: I moved it to the in person context

**Review by brianseek — APPROVED** [review]
Nice! Good to merge once un-blocked.

**Pull request merged**
